src/ssl.c

```diff
@@ -12946,6 +12947,26 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
         ctx->cm               = str->cm;
         ctx->x509_store.cm    = str->cm;
 
+        /* wolfSSL_CTX_set_cert_store() (this function) associates str with the
+         * wolfSSL_CTX. It is clear that this is a TLS use case which means we
+         * should move all the certs, if any, into the CertMgr and set
+         * str->certs to NULL as that will allow the certs to be properly
+         * processed. */
+        if (str->certs != NULL) {
+            while (wolfSSL_sk_X509_num(str->certs) > 0) {
+                int ret;
+                x = wolfSSL_sk_X509_pop(str->certs);
```

Comment:
The cert store has 3 X509 stacks, one for `certs`, one for `trusted`, and one named `owned`. Are we sure that it is the `certs` stack that should get added as CA's?